builtin/credential/cert/path_crls.go

```diff
@@ -188,44 +195,84 @@ func (b *backend) pathCRLWrite(ctx context.Context, req *logical.Request, d *fra
 	if name == "" {
 		return logical.ErrorResponse(`"name" parameter cannot be empty`), nil
 	}
-	crl := d.Get("crl").(string)
+	if crlRaw, ok := d.GetOk("crl"); ok {
+		crl := crlRaw.(string)
+		certList, err := x509.ParseCRL([]byte(crl))
+		if err != nil {
+			return logical.ErrorResponse(fmt.Sprintf("failed to parse CRL: %v", err)), nil
+		}
+		if certList == nil {
+			return logical.ErrorResponse("parsed CRL is nil"), nil
+		}
 
-	certList, err := x509.ParseCRL([]byte(crl))
-	if err != nil {
-		return logical.ErrorResponse(fmt.Sprintf("failed to parse CRL: %v", err)), nil
-	}
-	if certList == nil {
-		return logical.ErrorResponse("parsed CRL is nil"), nil
+		err = b.setCRL(ctx, req.Storage, certList, name, nil)
+		if err != nil {
+			return nil, err
+		}
+	} else if urlRaw, ok := d.GetOk("url"); ok {
+		url := urlRaw.(string)
+		if url == "" {
+			return logical.ErrorResponse("empty CRL url"), nil
+		}
+		_, err := url2.Parse(url)
+		if err != nil {
+			return logical.ErrorResponse("invalid CRL url: %v", err), nil
+		}
+		err = b.setCRL(ctx, req.Storage, nil, name, &CDPInfo{
```

Comment:
Should we not call b.fetchCRL before the call to b.setCRL? That would validate that the URL they provided us could be properly accessed and parsed before us updating the configuration? 

If I understand this current code, we will update the configuration on disk, then fail within b.updateCRLs call below returning an error back to the end-user but the configuration was updated.

Reply:
The problem is fetchCRL loads the config, which hasn't been written yet.  So we setCRL first sort of as a waste, and then follow up with fetchCRL.  You're right though this leaves it in a bad state.  I'll address.

Reply:
Oh, I see, simply doing fetchCRL should be enough, since it will subsequently setCRL too.